File: services/api_python/src/types/product.py
"""
Pydantic schemas for product catalogue endpoints.
"""

import uuid
from decimal import Decimal

from pydantic import BaseModel


class ProductListItem(BaseModel):
    """Single product in a catalogue listing."""

    id: uuid.UUID
    name: str
    price: Decimal
    thumbnail_url: str | None
    stock_quantity: int
    stock_status: str
    category_slug: str

    model_config = {"from_attributes": True}


class CategoryResponse(BaseModel):
    """Category in a listing."""

    id: uuid.UUID
    name: str
    slug: str

    model_config = {"from_attributes": True}


class PaginatedResponse(BaseModel):
    """Paginated wrapper for product listings."""

    items: list[ProductListItem]
    total: int
    page: int
    limit: int
    has_next: bool

File: frontend/dashboard/src/types/product.ts
export interface ProductListItem {
  id: string
  name: string
  price: number
  thumbnail_url: string | null
  stock_quantity: number
  stock_status: 'in_stock' | 'out_of_stock'
  category_slug: string
}

export interface Category {
  id: string
  name: string
  slug: string
}

export interface PaginatedResponse {
  items: ProductListItem[]
  total: number
  page: number
  limit: number
  has_next: boolean
}

export interface ProductDetail {
  id: string
  name: string
  description: string
  price: number
  stock_quantity: number
  stock_status: 'in_stock' | 'out_of_stock'
  thumbnail_url: string | null
  images: string[]
  category_name: string
  category_slug: string
}
